lib/manager/helmfile/extract.ts

```diff
@@ -0,0 +1,77 @@
+import is from '@sindresorhus/is';
+import yaml from 'js-yaml';
+
+import { logger } from '../../logger';
+import { PackageFile, PackageDependency, ExtractConfig } from '../common';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string,
+  config: ExtractConfig
+): PackageFile {
+  let deps = [];
+  let doc;
+  const aliases: Record<string, string> = {};
+  try {
+    doc = yaml.safeLoad(content, { json: true });
+  } catch (err) {
+    logger.debug({ fileName }, 'Failed to parse helmfile helmfile.yaml');
+    return null;
+  }
+  if (!(doc && is.array(doc.releases))) {
+    logger.debug({ fileName }, 'helmfile.yaml has no releases');
+    return null;
+  }
+
+  if (doc.repositories) {
+    for (let i = 0; i < doc.repositories.length; i += 1) {
+      aliases[doc.repositories[i].name] = doc.repositories[i].url;
+    }
+  }
+  logger.debug({ aliases }, 'repositories discovered.');
+
+  deps = doc.releases.map(dep => {
+    let depName = dep.chart;
+    let repoName = null;
+
+    if (dep.chart.startsWith('./')) {
+      return {
+        depName,
+        skipReason: 'local-chart',
+      } as PackageDependency;
+    }
+
+    if (dep.chart.includes('/')) {
+      const v = dep.chart.split('/');
+      depName = v[1];
+      repoName = v[0];
+    } else {
+      repoName = dep.chart;
+    }
+
+    const res: PackageDependency = {
+      depName,
+      currentValue: dep.version,
+      registryUrls: [aliases[repoName]]
+        .concat([config.aliases[repoName]])
+        .filter((v, _) => {
+          return v != null;
+        }),
+    };
+
+    if (res.depName.includes('{') || res.depName.includes('}')) {
+      res.skipReason = 'invalid-chart';
```

Comment:
Is it really invalid, or something type of placeholder?

Reply:
as far as I see in helmfile is `./` for local or `repo/chart` and the helm documentation is giving that definition of valid chart names:

```Chart names should be lower case letters and numbers. Words may be separated with dashes (-)```

Probably will also be possible to check for all special character because helmfile is preventing you to use `@` placeholder (like in helm):

```n ./helmfile.yaml: failed to read helmfile.yaml: reading document at index 1: yaml: line 19: found character that cannot start any token```

line 19 is `chart: @stable/nginx-ingress`

In the specific cases `{}` are used by helmfile to render template stuff based on go and the chart name is generated using some internals of helmfile.

Reply:
Maybe instead of invalid-chart we have a term like unsupported-chart-type or something like that? I mean if it's invalid, but we can't update it, then we don't need to say to the user it's invalid or they may report it as a bug.